scripts/visualize/visualize_utils.R

```diff
@@ -169,9 +169,53 @@ set_sp_plot <- function(){
   par(mai=c(0,0,0,0), omi=c(0,0,0,0), xaxs = 'i', yaxs = 'i')
 }
 
+locate_css_class <- function(css, class_nm){
+  start_class <- grep(class_nm, css)
+  close_classes <- grep("}", css)
+  end_class <- close_classes[which(start_class < close_classes)[1]]
+  css_details <- css[start_class:end_class]
+  return(css_details)
+}
+
+locate_css_class_detail <- function(css, item_nm){
+  item_line <- grep(item_nm, css)
+  strip_excess1 <- gsub(paste0("  ", item_nm, ": "), "", css[item_line])
+  item_details <- gsub(";", "", strip_excess1)
+  return(item_details)
+}
+
 #' script to turn the dataviz into a thumbnail
 #' 
 visualize.map_thumbnail <- function(viz){
-  data <- readDepends(viz)
-  file.copy(from = data[[1]], to = viz[['location']], overwrite = TRUE)
+  depends <- readDepends(viz)
+  states <- depends[[1]]
+  islands <- depends[[2]]
+  counties <- depends[[3]]
+  precip <- depends[[4]]
+
+  # styling details
+  css <- readLines(viz$css)
```

Comment:
this is going to be fragile (if we can things in css), but I don't see a better way under the time constraints

Reply:
look for a package or something, but couldn't find one